### ISSUE TITLE
Can you freeze and add multiple files?

### ISSUE THREAD

**Issue by AlasdairAtKitronik** [mytechnotalent/MicroPython-micro-bit_How_To_Build_And_Freeze_Modules#1]
First off, thank you for putting this together - it's very helpfully solved a problem I was working through a couple of weeks ago!

This isn't really an issue, more just a question: Following your steps I'm able to successfully freeze a single python file, build the firmware with it included and then use the 'module' in a program. However, it doesn't work when trying to add multiple frozen files to the firmware build - do you know if this is possible, and how it might be done? Thanks.

**Comment by mytechnotalent** [comment]
Hi @AlasdairAtKitronik great question! You can add one file at a time.  What you want to do is delete the original one after you have created and moved your .c file and replace it with a new .py file and repeat the process.

**Comment by AlasdairAtKitronik** [comment]
Thanks, that's worked now. The only thing to note that is different, is that the three variable names need to be different in the different .c files, otherwise there is a 'this is already defined error'. For my second file to add, I just added '2' to the end of each one :

'''blocks
mp_frozen_str_names -> mp_frozen_str_names2 
mp_frozen_str_sizes -> mp_frozen_str_sizes2
mp_frozen_str_content -> mp_frozen_str_content2
```

**Comment by mytechnotalent** [comment]
Oh very good point @AlasdairAtKitronik!  I think however if you delete the build directory before hand and re-build it should work without this.  Can you give that a try and let me know?

**Comment by AlasdairAtKitronik** [comment]
I deleted the build directory and also did a 'yt clean' as well. Still getting the same "multiple definition of <mp_frozen_variable>" errors and it fails.

**Comment by mytechnotalent** [comment]
Can you fork the repo here and do a PR and update the README.md with your new steps?

**Comment by AlasdairAtKitronik** [comment]
Of course, I'll do that now.

**Comment by mytechnotalent** [comment]
I so appreciate that.  I would LOVE if you would help me and be a partner on my other repo that more formally works with this https://github.com/mytechnotalent/minipython.

Would you be open to forking and joining me and updating that documentation (for the 2.1 release) and perhaps help me build out that project after you update this one?

https://github.com/mytechnotalent/minipython is my attempt to help non-technical educators learn and teach Python to their students and right now its only me working the repo.

**Comment by mytechnotalent** [comment]
Regarding the other repo here is where if you can do a PR to update on the develop branch - https://minipython.readthedocs.io/en/latest/tutorials/modules.html.

This again is in the https://github.com/mytechnotalent/minipython repo.

**Comment by AlasdairAtKitronik** [comment]
Pull request created.
With regards to your other repo, I might be able to do it in my own time - but currently this is a work account. Let me have a think. Thank you for the offer though.

**Comment by mytechnotalent** [comment]
Thank you that would be great I could really use the help on such an important cause of helping non-technical educators learning Python.

**Comment by AlasdairAtKitronik** [comment]
I'm very sorry, but it appears that there is still an issue witht the multiple files - renaming the variables means that the 'make all' command completes succesfully and builds the firmware, however, only the module with the original variable names actually appears when the modules are searched using:
```python
help ('modules')
```
I think it might be best to revert that pull request temporarily while I have a look at this.

**Comment by mytechnotalent** [comment]
No worries I can just leave it and you can do another PR when you have time.  I so appreciate the help!

**Comment by AlasdairAtKitronik** [comment]
Sorry for going quiet on this. 
I have spent a long time trying to get round the issue, but have not found a solution at all for being able to freeze multiple modules. The issue seems to be that in 'frozenmod.c', the file carrying out the actual freezing, the function is looking for the specific variable names (mp_frozen_str_names[], mp_frozen_str_sizes[] & mp_frozen_str_content[]). I tried changing where these are declared to 'frozenmod.h', and then including this header in the generated frozen py file, but I kept getting 'multiple definition' errors. 
I'm afraid this is beyond me at the moment. I would recommend removing the instructions I added for freezing multiple files.

**Comment by mytechnotalent** [comment]
What happens when you

**Comment by mytechnotalent** [comment]
What happens when you remove the old .py files completely and keep only the main .py file?  What I mean is you freeze a module, put the .c file in the new directory and delete the old .py file and do another .py file, freeze and move the .c file to the proper directory and repeat?

**Comment by AlasdairAtKitronik** [comment]
That's what I have been doing. Each frozen file in the .c format has the same variable names though.
I tried rewriting the frozenmod,c function to look for different variable names (like my initial instructions said), but that kept failing as well. 
The only solution I've found is merging all the .py files into one big .py file, and then freezing that single one. Doesn't feel like an elegant solution though.
